### PR TITLE
Add unit tests for peerSignatureHandler

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
Added unit tests for peerSignatureHandler.

### CODE REVIEW

File: crypto/mock/cacherMock.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
+package mock
```
Comment (sasurobert) [inline crypto/mock/cacherMock.go:1]: you can take this from testcommon package. like use it. no need to redefine

File: crypto/mock/cacherMock.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	cacher.mutAddedDataHandlers.RLock()
+	for _, handler := range cacher.addedDataHandlers {
+		go handler(key, val)
+	}
+	cacher.mutAddedDataHandlers.RUnlock()
+}
+
+// Get -
+func (cacher *CacherMock) Get(key []byte) (value interface{}, ok bool) {
+	cacher.mut.Lock()
```
Comment (SebastianMarian) [inline crypto/mock/cacherMock.go:52]: RLock/RUnlock

File: crypto/mock/cacherMock.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	defer cacher.mut.Unlock()
+
+	val, ok := cacher.dataMap[string(key)]
+
+	return val, ok
+}
+
+// Has -
+func (cacher *CacherMock) Has(key []byte) bool {
+	cacher.mut.Lock()
```
Comment (SebastianMarian) [inline crypto/mock/cacherMock.go:62]: RLock/RUnlock

File: crypto/mock/cacherMock.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	defer cacher.mut.Unlock()
+
+	_, ok := cacher.dataMap[string(key)]
+
+	return ok
+}
+
+// Peek -
+func (cacher *CacherMock) Peek(key []byte) (value interface{}, ok bool) {
+	cacher.mut.Lock()
```
Comment (SebastianMarian) [inline crypto/mock/cacherMock.go:72]: RLock/RUnlock

File: crypto/mock/cacherMock.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+		keys[idx] = []byte(k)
+		idx++
+	}
+
+	return keys
+}
+
+// Len -
+func (cacher *CacherMock) Len() int {
+	cacher.mut.Lock()
```
Comment (SebastianMarian) [inline crypto/mock/cacherMock.go:117]: RLock/RUnlock

File: crypto/mock/cacherMock.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply crypto/mock/cacherMock.go:1]: Done.

File: crypto/mock/cacherMock.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply crypto/mock/cacherMock.go:52]: Refactored.

File: crypto/mock/cacherMock.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply crypto/mock/cacherMock.go:62]: Refactored.

File: crypto/mock/cacherMock.go (hunk 4 above)
Comment (BeniaminDrasovean) [reply crypto/mock/cacherMock.go:72]: Refactored.

File: crypto/mock/cacherMock.go (hunk 5 above)
Comment (BeniaminDrasovean) [reply crypto/mock/cacherMock.go:117]: Refactored.

**Pull request merged**